pkg/types/types.go

```diff
@@ -23,67 +23,136 @@ import (
 	"github.com/project-oak/transparent-release/pkg/intoto"
 )
 
-// ValidatedProvenance wraps an intoto.Statement representing a valid SLSA provenance statement.
-// A provenance statement is valid if it contains a single subject, with a SHA256 hash.
-type ValidatedProvenance struct {
+// ProvenanceIR wraps an intoto.Statement representing a valid SLSA provenance statement. A provenance statement is valid if it contains a single subject, with a SHA256 hash.
+// ProvenanceIR also holds internal intermediate representations of data from provenances. We want to map different provenances of different build types to ProvenanceIR, so
+// all fields except for `provenanceStatement` are optional.
+type ProvenanceIR struct {
 	// The field is private so that invalid instances cannot be created.
-	provenance intoto.Statement
+	provenanceStatement      *intoto.Statement
```

Comment:
Why do we still need to keep the provenance statement? It feels redundant.

Reply:
Update after discussion: We need the `intoto.Statement` ih the `ReproducibleProvenanceVerifier` for [LoadBuildConfig](https://github.com/project-oak/transparent-release/blob/a765fb136047cb4d6034012e3d91d76053705a70/internal/common/common.go#L129)